app/src/editor/canvas/index.jsx

```diff
@@ -160,13 +160,27 @@ const CanvasEditComponent = class CanvasEdit extends Component {
         ))
     }
 
-    addModule = async ({ id, configuration }) => {
+    addModule = async ({ id, configuration, position }) => {
         const action = { type: 'Add Module' }
         const moduleData = await sharedServices.getModule({
             id,
             configuration,
         })
         if (this.unmounted) { return }
+
+        // Overwrite layout position if it was provided
+        if (position) {
+            const newPosition = {
+                left: `${position.x}px`,
+                top: `${position.y}px`,
+            }
+
+            moduleData.layout = {
+                ...moduleData.layout,
+                position: newPosition,
+            }
+        }
```

Comment:
No need, just drop it into `configuration`:
```js
        const moduleData = await sharedServices.getModule({
            id,
            configuration: {
                ...configuration,
                layout: {
                    ...configuration.layout,
                    position: {
                        left: `${position.x}px`,
                        top: `${position.y}px`,
                    }
                }
            }
        })
```

Also thinking perhaps this embedding into `configuration.layout` should happen at the canvas `addModule` callsite to keep the less convenient but "standard" `layout` format? Then this doesn't need to know anything about a custom `position` and only deals with module-formatted data.